src/tools.rs

```diff
@@ -243,6 +239,15 @@ pub fn guess_language<P: AsRef<Path>>(buf: &[u8], path: P) -> (Option<LANG>, Str
     }
 }
 
+pub(crate) fn remove_blank_lines(data: &mut Vec<u8>) {
```

Comment:
I'd say that 99.9% of the time the file has at least one trailing EOL. So I think we should compute `count_trailing` and then if !0 then `truncate` (which is better than resize) else push an EOL.